carma-messenger-core/cpp_message/src/SDSM_Message.cpp

```diff
@@ -1072,4 +1072,958 @@ namespace cpp_message
+                long semi_minor = sdsm_core.refPosXYConf.semiMinor;
+                if(semi_minor > j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_MAX){
+                    RCLCPP_WARN_STREAM(node_logging_->get_logger(), "Decoded SDSM semiMinor above max, setting to max");
+                    semi_minor = j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_MAX;
+                }
+                else if(semi_minor < j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_MIN){
+                    RCLCPP_WARN_STREAM(node_logging_->get_logger(), "Decoded SDSM semiMinor below min, setting to min");
+                    semi_minor = j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_MIN;
+                }
+                output.ref_pos_xy_conf.semi_minor = semi_minor;
+            }
+            else{
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(), "Decoded SDSM missing semiMinor, set to unavailable");
+                output.ref_pos_xy_conf.semi_minor = j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE;
+            }
+            // PositionalAccuracy - orientation
+            if(sdsm_core.refPosXYConf.orientation && sdsm_core.refPosXYConf.orientation != j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_UNAVAILABLE){
+                long orientation = sdsm_core.refPosXYConf.orientation;
+                if(orientation > j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_MAX){
+                    RCLCPP_WARN_STREAM(node_logging_->get_logger(), "Decoded SDSM orientation above max, setting to max");
+                    orientation = j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_MAX;
+                }
+                else if(orientation < j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_MIN){
+                    RCLCPP_WARN_STREAM(node_logging_->get_logger(), "Decoded SDSM orientation below min, setting to min");
+                    orientation = j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_MIN;
+                }
+                output.ref_pos_xy_conf.orientation = orientation;
+            }
+            else{
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(), "Decoded SDSM missing orientation, set to unavailable");
+                output.ref_pos_xy_conf.orientation = j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_UNAVAILABLE;  
+            }
+
+            // ElevationConfidence
+            if(sdsm_core.refPosElConf){
+                output.presence_vector |= j3224_v2x_msgs::msg::SensorDataSharingMessage::HAS_REF_POS_EL_CONF;
+                output.ref_pos_el_conf.confidence = *sdsm_core.refPosElConf;
+            }
+
+            // objects
+            j3224_v2x_msgs::msg::DetectedObjectList detected_objects;
+
+
+            for(auto obj_itr = 0; obj_itr < sdsm_core.objects.list.count; ++obj_itr){
```

Comment:
Suspicious `for` loop. Can we use a range-based `for` loop or use iterators?

Reply:
After some internal trial and error, we determined this would be more effort than it's worth due to the ASN.1 types.